### DISCUSSION TITLE
Neural network and survival analysis

### DISCUSSION THREAD

**Discussion by Pina1804** [Lightning-AI/dl-fundamentals#109] [Exercises]
want to predict the risk of having a heart attack using deep surv. I have a dataset where three cases are possible: a patient has had only one heart attack, so there is only one row for them; a patient has had two heart attacks, so there are two corresponding rows for them; a patient has not had any heart attacks, so there is one row corresponding to them. 
Here's your provided code:

```python
f = df.sort_values(by=['ID', 'Time'])
unique_ids = df['ID'].unique()
ids_train, ids_test = train_test_split(unique_ids, test_size=0.2, random_state=123)
ids_train, ids_val = train_test_split(ids_train, test_size=0.2, random_state=123)
df_train = df[df['ID'].isin(ids_train)]
df_val = df[df['ID'].isin(ids_val)]
df_test = df[df['ID'].isin(ids_test)]
ssc = StandardScaler()
x_train = ssc.fit_transform(df_train.drop(columns=['Duration_Years', 'Evento', 'Time'])).astype('float32')
x_val = ssc.transform(df_val.drop(columns=['Duration_Years', 'Evento', 'Time'])).astype('float32')
x_test = ssc.transform(df_test.drop(columns=['Duration_Years', 'Evento', 'Time'])).astype('float32')
get_target = lambda df: (df['Duration_Years'].values, df['Evento'].values)
y_train = get_target(df_train)
y_val = get_target(df_val)
durations_test, events_test = get_target(df_test)
val = x_val, y_val
in_features = x_train.shape[1]
num_nodes = [32, 32]
out_features = 1
batch_norm = True
dropout = 0.1
output_bias = False
net = tt.practical.MLPVanilla(in_features, num_nodes, out_features, batch_norm,
                              dropout, output_bias=output_bias)


model = CoxPH(net, tt.optim.Adam)
batch_size = 256
lrfinder = model.lr_finder(x_train, y_train, batch_size, tolerance=10)
_ = lrfinder.plot()
lrfinder.get_best_lr()
model.optimizer.set_lr(0.1)
epochs = 512
callbacks = [tt.callbacks.EarlyStopping()]
verbose = True
log = model.fit(x_train, y_train, batch_size, epochs, callbacks, verbose,
                val_data=val, val_batch_size=batch_size)
_ = log.plot()
```

Is this code correct for what I want to do, and more importantly, can the algorithm understand when there are multiple rows for the same patient? please help me

**Comment by rasbt** [comment]
Hi there, it seems to me that what you are trying to predict is an ordinal variable, i.e., you have an order 0 < 1 < 2. For this, you might actually find my CORN and CORAL methods for ordinal regression with deep neural networks helpful: https://github.com/Raschka-research-group/coral-pytorch

E.g. here I have an example with PyTorch Lightning that might be useful: https://github.com/Raschka-research-group/coral-pytorch/blob/main/docs/tutorials/pytorch_lightning/ordinal-corn_cement.ipynb
